Prepare Detekt 1.19.0-RC1 (#4215)
detekt/detekt: docs/pages/changelog 1.x.x.md

toc: true
---

#### 1.19.0-RC1 - 2021-10-31

##### Notable Changes

> - We now offer a `ignoreAnnotated` configuration key that you can use on all your rules to suppress findings if inside an annotated block (e.g. `@Composable`) - [#4102](https://github.com/detekt/detekt/pull/4102)

Reviewer: ```suggestion
- We now offer an `ignoreAnnotated` configuration key that you can use on all your rules to suppress findings if inside an annotated block (e.g. `@Composable`) - [#4102](https://github.com/detekt/detekt/pull/4102)
```

- Report configuration is changing in the Gradle plugin. The `reports` extension on the `detekt` extension has been
  deprecated. See the Migration section below for steps to migrate to the new recommended configuration - [#3687](https://github.com/detekt/detekt/pull/3687)
- The `ExplicitCollectionElementAccessMethod` rule is now a type-resolution only rule - [#4201](https://github.com/detekt/detekt/pull/4201)

> - We bumped KtLint and updated the default rule you get enabled to mirror what KtLint is doing - [#4179](https://github.com/detekt/detekt/pull/4179)

Reviewer: ```suggestion
- We bumped ktlint and updated the default enabled rules to mirror what ktlint is doing - [#4179](https://github.com/detekt/detekt/pull/4179)
```

- Add a new ConsoleReport format - [#4027](https://github.com/detekt/detekt/pull/4027)
- Gradle: We removed the `afterEvaluate` wrapper from the Android and KMM plugin - [#4159](https://github.com/detekt/detekt/pull/4159)

> - We now test against Java 17 - [#4136](https://github.com/detekt/detekt/pull/4136)

Reviewer: ```suggestion
- We now test against Java 17 and stopped testing against Java 16 - [#4136](https://github.com/detekt/detekt/pull/4136)
```

}
```

##### Changelog

- ForbiddenMethodCall: report overriding method calls - [#4205](https://github.com/detekt/detekt/pull/4205)
- ObjectLiteralToLambda: fix false positive when using Java interfaces with default methods - [#4203](https://github.com/detekt/detekt/pull/4203)
- Unit tests for TooGenericExceptionThrown - [#4198](https://github.com/detekt/detekt/pull/4198)
- Display correct --jvm-target values when using --help flag - [#4195](https://github.com/detekt/detekt/pull/4195)
- Improved `MaximumLineLength` documentation - [#4188](https://github.com/detekt/detekt/pull/4188)
- Report NewLineAtEndOfFile source location at end of file - [#4187](https://github.com/detekt/detekt/pull/4187)
- #4169 OutdatedDocumentation rule - [#4185](https://github.com/detekt/detekt/pull/4185)
- Don't report on platform types in NullableToStringCall - [#4180](https://github.com/detekt/detekt/pull/4180)
- Fix #4140: Allow Bazel based tests to run with string test input - [#4170](https://github.com/detekt/detekt/pull/4170)

> - Improve ForbiddenMethodCall documenation - [#4166](https://github.com/detekt/detekt/pull/4166)

Reviewer: ```suggestion
- Improve ForbiddenMethodCall documentation - [#4166](https://github.com/detekt/detekt/pull/4166)
```

- Report SwallowedException on catchParameter - [#4158](https://github.com/detekt/detekt/pull/4158)
- Enable binary compatibility validator for detekt-test and detekt-test-api - [#4157](https://github.com/detekt/detekt/pull/4157)
- Fix issues with Elvis operator in UnconditionalJumpStatementInLoop - [#4150](https://github.com/detekt/detekt/pull/4150)

> - Split InvalidPackageDeclaration to create MissingPackageDeclaration - [#4149](https://github.com/detekt/detekt/pull/4149)

Reviewer: Notable change?

- Improve documentation for naming rules - [#4146](https://github.com/detekt/detekt/pull/4146)

> - Add customMessage to ForbiddenComment Rule - [#4126](https://github.com/detekt/detekt/pull/4126)

Reviewer: Notable change?

- Use Codecov GitHub Action to upload coverage - [#4184](https://github.com/detekt/detekt/pull/4184)
- Enable ParameterListWrapping rule on detekt codebase - [#4178](https://github.com/detekt/detekt/pull/4178)
- Add test cases for MagicNumber - [#4152](https://github.com/detekt/detekt/pull/4152)
- Fix FunctionParameterNamingSpec - [#4145](https://github.com/detekt/detekt/pull/4145)
- Address feedback on #4139 - [#4143](https://github.com/detekt/detekt/pull/4143)
- Don't skip tests that now pass - [#4142](https://github.com/detekt/detekt/pull/4142)
- Fixes for Kotlin 1.6.0-M1 - [#4139](https://github.com/detekt/detekt/pull/4139)
- Don't unnecessarily propogate opt-in requirement - [#4116](https://github.com/detekt/detekt/pull/4116)
- Drop junit-platform-launcher dependency - [#4115](https://github.com/detekt/detekt/pull/4115)
- Ensure detekt-tooling public API is stable - [#4112](https://github.com/detekt/detekt/pull/4112)
- Fix globing typo - [#4107](https://github.com/detekt/detekt/pull/4107)
- Rename and split ValidateConfig files - [#4105](https://github.com/detekt/detekt/pull/4105)
- Dynamic deprecation - [#4104](https://github.com/detekt/detekt/pull/4104)

> - Fix indent issues with continaution indent - [#4103](https://github.com/detekt/detekt/pull/4103)

Reviewer: ```suggestion
- Fix indent issues with continuation indent - [#4103](https://github.com/detekt/detekt/pull/4103)
```
